pysteps/nowcasts/linda.py

```diff
@@ -0,0 +1,1407 @@
+            )
+            p_opt = _get_anisotropic_kernel_params(p_opt.x)
+
+            return _compute_kernel_anisotropic(p_opt, **kernel_params)
+        else:
+            p_opt = opt.minimize_scalar(
+                objf_iso, bounds=[0.01, 10.0], method="bounded", args=(i,)
+            )
+            p_opt = p_opt.x
+
+            return _compute_kernel_isotropic(p_opt, **kernel_params)
+
+    if DASK_IMPORTED and num_workers > 1:
+        res = []
+        for i in range(len(weights)):
+            res.append(dask.delayed(worker)(i))
+        kernels = dask.compute(*res, num_workers=num_workers, scheduler="threads")
+    else:
+        kernels = []
+        for i in range(len(weights)):
+            kernels.append(worker(i))
+
+    return kernels
+
+
+def _estimate_perturbation_params(
+    fct_err,
+    fct_gen,
+    errdist_window_radius,
+    acf_window_radius,
+    interp_window_radius,
+    measure_time,
+    num_workers,
+):
+    """Estimate perturbation generator parameters from forecast errors."""
+    pert_gen = {}
+    pert_gen["m"] = fct_err.shape[0]
+    pert_gen["n"] = fct_err.shape[1]
+
+    feature_coords = fct_gen["feature_coords"]
+
+    print("Estimating perturbation parameters... ", end="", flush=True)
+
+    if measure_time:
+        starttime = time.time()
+
+    mask_finite = np.isfinite(fct_err)
+
+    fct_err = fct_err.copy()
+    fct_err[~mask_finite] = 1.0
+
+    weights_dist = _compute_window_weights(
+        feature_coords,
+        fct_err.shape[0],
+        fct_err.shape[1],
+        errdist_window_radius,
+    )
+
+    acf_winfunc = _window_tukey if feature_coords.shape[0] > 1 else _window_uniform
+
+    def worker(i):
+        weights_acf = acf_winfunc(
+            fct_err.shape[0],
+            fct_err.shape[1],
+            feature_coords[i, 0],
+            feature_coords[i, 1],
+            acf_window_radius,
+            acf_window_radius,
+        )
+
+        mask = np.logical_and(mask_finite, weights_dist[i] > 0.1)
+        if np.sum(mask) > 10 and np.sum(np.abs(fct_err[mask] - 1.0) >= 1e-3) > 10:
+            distpar = _fit_dist(fct_err, stats.lognorm, weights_dist[i], mask)
+            inv_acf_mapping = _compute_inverse_acf_mapping(stats.lognorm, distpar)
+            mask_acf = weights_acf > 1e-4
+            std = _weighted_std(fct_err[mask_acf], weights_dist[i][mask_acf])
+            acf = inv_acf_mapping(
+                _compute_sample_acf(weights_acf * (fct_err - 1.0) / std)
+            )
+            acf = _fit_acf(acf)
+        else:
+            distpar = None
+            std = None
+            acf = None
+
+        return distpar, std, np.sqrt(np.abs(np.fft.rfft2(acf)))
+
+    dist_params = []
+    stds = []
+    acf_fft_ampl = []
+
+    if DASK_IMPORTED and num_workers > 1:
+        res = []
+        for i in range(feature_coords.shape[0]):
+            res.append(dask.delayed(worker)(i))
+        res = dask.compute(*res, num_workers=num_workers, scheduler="multiprocessing")
```

Comment:
can you explain why you selected "multiprocessing" here instead of "threads" as elsewhere? 
For some reasons, this is giving me a runtime error:
```
RuntimeError:
        An attempt has been made to start a new process before the
        current process has finished its bootstrapping phase.
```

Reply:
The reason is that "threads" gives absolutely no parallelization here. This is probably because of the GIL. I agree that "multiprocessing" is not the best solution, but so far I haven't figured out a better way.

Reply:
Thanks @dnerini for your effort so far. You have done quite extensive work to go through this pull request.

Reply:
You're right about the GIL, although I did experience a significant speedup with "threads" (not sure why and how).

But my main question here is why the runtime error? is it only me or there is a bug somewhere we need to fix before releasing it? would be safer to set it to "threads" instead?

Reply:
I think I found the problem. For some reason, all programs that call a module containing `dask.compute(...,scheduler="multiprocessing")` need to do this inside the `if __name__ == "__main__":` block. This is not very convenient for the user. So let's use "threads" instead.